File: Aruvi21042026/Aruvi21042026/Arivi App Final/app/src/main/java/com/Saalai/SalaiMusicApp/Models/PlaylistSection.java
package com.Saalai.SalaiMusicApp.Models;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;
import java.io.Serializable;

public class PlaylistSection implements Serializable {

    @SerializedName("sectionId")
    private String sectionId;

    @SerializedName("sectionTitle")
    private String sectionTitle;

    @SerializedName("categories")  // ← CRITICAL: Maps JSON "categories" to this field
    private List<ArtistCategory> artistCategories;

    @SerializedName("layoutType")
    private int layoutType;

    @SerializedName("spanCount")
    private int spanCount;

    // Constructors
    public PlaylistSection() {
        this.spanCount = 2;
    }

    public PlaylistSection(String sectionTitle, List<ArtistCategory> artistCategories, int layoutType) {
        this();
        this.sectionTitle = sectionTitle;
        this.artistCategories = artistCategories;
        this.layoutType = layoutType;
    }

    public PlaylistSection(String sectionTitle, List<ArtistCategory> artistCategories) {
        this(sectionTitle, artistCategories, 1);
    }

    public PlaylistSection(String sectionTitle, List<ArtistCategory> artistCategories, int layoutType, int spanCount) {
        this();
        this.sectionTitle = sectionTitle;
        this.artistCategories = artistCategories;
        this.layoutType = layoutType;
        this.spanCount = spanCount;
    }

    public PlaylistSection(String sectionId, String sectionTitle, List<ArtistCategory> artistCategories, int layoutType, int spanCount) {
        this.sectionId = sectionId;
        this.sectionTitle = sectionTitle;
        this.artistCategories = artistCategories;
        this.layoutType = layoutType;
        this.spanCount = spanCount;
    }

    // Getters and Setters
    public String getSectionId() {
        return sectionId;
    }

    public void setSectionId(String sectionId) {
        this.sectionId = sectionId;
    }

    public String getSectionTitle() {
        return sectionTitle;
    }

    public void setSectionTitle(String sectionTitle) {
        this.sectionTitle = sectionTitle;
    }

    public String getSectionName() {
        return sectionTitle;
    }

    public void setSectionName(String sectionName) {
        this.sectionTitle = sectionName;
    }

    public List<ArtistCategory> getArtistCategories() {
        if (artistCategories == null) {
            artistCategories = new ArrayList<>();
        }
        return artistCategories;
    }

    public void setArtistCategories(List<ArtistCategory> artistCategories) {
        this.artistCategories = artistCategories;
    }

    public List<ArtistCategory> getArtists() {
        return getArtistCategories();
    }

    public void setArtists(List<ArtistCategory> artists) {
        this.artistCategories = artists;
    }

    public int getLayoutType() {
        return layoutType;
    }

    public void setLayoutType(int layoutType) {
        this.layoutType = layoutType;
    }

    public int getSpanCount() {
        return spanCount;
    }

    public void setSpanCount(int spanCount) {
        this.spanCount = spanCount;
    }

    @Override
    public String toString() {
        return "PlaylistSection{" +
                "sectionId='" + sectionId + '\'' +
                ", sectionTitle='" + sectionTitle + '\'' +
                ", artistCategoriesCount=" + (artistCategories != null ? artistCategories.size() : 0) +
                ", layoutType=" + layoutType +
                ", spanCount=" + spanCount +
                '}';
    }
}

File: aruviflutter/lib/models/playlist_section.dart
import 'artist_category.dart';

class PlaylistSection {
  String? sectionId;
  String? sectionTitle;
  List<ArtistCategory> categories;
  int layoutType;
  int spanCount;

  PlaylistSection({
    this.sectionId,
    this.sectionTitle,
    this.categories = const [],
    this.layoutType = 1,
    this.spanCount = 2,
  });

  factory PlaylistSection.fromJson(Map<String, dynamic> json) {
    var categoriesList = json['categories'] as List? ?? [];
    List<ArtistCategory> parsedCategories = categoriesList.map((c) => ArtistCategory.fromJson(c)).toList();

    return PlaylistSection(
      sectionId: json['sectionId']?.toString(),
      sectionTitle: json['sectionTitle']?.toString() ?? json['sectionName']?.toString(),
      layoutType: json['layoutType'] != null ? int.tryParse(json['layoutType'].toString()) ?? 1 : 1,
      spanCount: json['spanCount'] != null ? int.tryParse(json['spanCount'].toString()) ?? 2 : 2,
      categories: parsedCategories,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'sectionId': sectionId,
      'sectionTitle': sectionTitle,
      'layoutType': layoutType,
      'spanCount': spanCount,
      'categories': categories.map((c) => c.toJson()).toList(),
    };
  }
}
